src/zip_node.js

```diff
@@ -1,16 +1,73 @@
 const { Buffer } = require('buffer')
 const fs = require('fs')
-const { basename, extname, normalize, sep } = require('path')
+const os = require('os')
+const { basename, extname, join, normalize, sep } = require('path')
 const { promisify } = require('util')
 
+const cpFile = require('cp-file')
+const makeDir = require('make-dir')
+const pMap = require('p-map')
 const unixify = require('unixify')
 
 const { startZip, addZipFile, addZipContent, endZip } = require('./archive')
+const { ARCHIVE_FORMAT_ZIP } = require('./utils/consts')
 
 const pStat = promisify(fs.stat)
+const pWriteFile = promisify(fs.writeFile)
 
-// Zip a Node.js function file
-const zipNodeJs = async function ({ basePath, destPath, filename, mainFile, pluginsModulesPath, aliases, srcFiles }) {
+// Taken from https://www.npmjs.com/package/cpy.
+const COPY_FILE_CONCURRENCY = os.cpus().length === 0 ? 2 : os.cpus().length * 2
+
+const createDirectory = async function ({
+  aliases = {},
+  basePath,
+  destFolder,
+  extension,
+  filename,
+  mainFile,
+  pluginsModulesPath,
+  srcFiles,
+}) {
+  const { contents: entryContents, filename: entryFilename } = getEntryFile({
+    commonPrefix: basePath,
+    filename,
+    mainFile,
+  })
+  const functionFolder = join(destFolder, basename(filename, extension))
+
+  // Creating the function folder.
+  await makeDir(functionFolder)
+
+  // Writing entry file.
+  await pWriteFile(join(functionFolder, entryFilename), entryContents)
+
+  // Copying source files.
+  await pMap(
```

Comment:
Would [`p-limit`](https://github.com/sindresorhus/p-limit) be better here, since we don't use the return value of `p-map`?